src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureMigrationEngineBuilder.cs

```diff
@@ -144,13 +144,28 @@ private string NormalizeAzureLink(string href, string defaultExtension, IMarkdow
         /// <param name="defaultExtension">default extension to append</param>
         /// <param name="isHrefRelativeNonMdFile">true if it is a relative path and not a markdown file. Otherwise false</param>
         /// <returns>Href with default extension appended</returns>
-        private string AppendDefaultExtension(string href, string defaultExtension, out bool isHrefRelativeNonMdFile)
+        private string AppendDefaultExtension(string href, string defaultExtension, IMarkdownContext context, string line, out bool isHrefRelativeNonMdFile)
         {
             isHrefRelativeNonMdFile = false;
-            if (!PathUtility.IsRelativePath(href))
+            // If the context doesn't have necessary info, return the original href
+            if (!context.Variables.ContainsKey("path"))
             {
                 return href;
             }
+            var currentFilePath = (string)context.Variables["path"];
+
+            try
+            {
+                if (!PathUtility.IsRelativePath(href))
+                {
+                    return href;
+                }
+            }
+            catch (ArgumentException)
```

Comment:
We can check the argument in the method. However, I still need the warning. Otherwise, user don't know this is a broken link.

---
In reply to: [75428301](https://github.com/dotnet/docfx/pull/620#discussion_r75428301) [](ancestors = 75428301,75424703,75423422,75423027)